src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -967,23 +968,19 @@ object Utils extends Logging {
     val outputStream = destFS.create(destPath)
 
     try {
-      var i = 0
-      while (i < srcFileStatuses.length) {
-        val fileStatus = srcFileStatuses(i)
-        val lenAdjust: Long = if (isBGzip && i < srcFileStatuses.length - 1)
-          -28
-        else
-          0
```

Comment:
@cseed, no, BGZFCodec doesn't write empty gzip blocks at the end of files. See 

https://github.com/HadoopGenomics/Hadoop-BAM/blob/master/src/main/java/org/seqdoop/hadoop_bam/util/BGZFCompressionOutputStream.java#L44